src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -169,14 +171,115 @@ private function createRelationsForItem($item, $formattedRelations)
                 }
             } elseif ($relation instanceof HasOne || $relation instanceof MorphOne) {
                 $modelInstance = $relation->updateOrCreate([], $relationDetails['values']);
+            } elseif ($relation instanceof HasMany || $relation instanceof MorphMany) {
+                $relation_values = $relationDetails['values'][$relationMethod];
+                // if relation values are null we can only attach, also we check if we sent a single dimensional array [1,2,3], or an array of arrays: [[1][2][3]]
+                // if is as single dimensional array we can only attach.
+                if ($relation_values === null || count($relation_values) == count($relation_values, COUNT_RECURSIVE)) {
+                    $this->attachManyRelation($item, $relation, $relationMethod, $relationDetails, $relation_values);
+                } else {
+                    $this->createManyEntries($item, $relation, $relationMethod, $relationDetails);
+                }
             }
-
             if (isset($relationDetails['relations'])) {
                 $this->createRelationsForItem($modelInstance, ['relations' => $relationDetails['relations']]);
             }
         }
     }
 
+    /**
+     * When using the HasMany/MorphMany relations as selectable elements we use this function to "mimic-sync" in those relations.
+     * Since HasMany/MorphMany does not have the `sync` method, we manually re-create it.
+     * Here we add the entries that developer added and remove the ones that are not in the list.
+     * This removal process happens with the following rules:
+     * - by default Backpack will behave like a `sync` from M-M relations: it deletes previous entries and add only the current ones.
+     * - `force_delete` is configurable in the field, it's `true` by default. When false, if connecting column is nullable instead of deleting the row we set the column to null.
```

Comment:
Are we _sure_ we want to `force_delete` by default? I imagine this scenario. I have a tag with A BUNCH of articles. One Tag HasMany Articles right? 

I go to the tag EDIT operation. This PR will make it so I have a new input there, which will allow me to select the Articles for this Tag, correct. Do I understand correctly that when I remove an Article association from a Tag... that Article will be deleted?

If so... I don't think that's the expected behavior...

Or have I greatly misunderstood this new functionality? 👀

Reply:
Hey @tabacitu , the relation setup is the following:

HasMany <-> BelongsTo

So, you have an Article -> HasMany -> Tags | Tag -> BelongsTo -> Article (via article_id in tags table).

If you are editing an Article you have a select for tags. When those tags get "unselected" they get deleted. If `force_delete = false` and column allows nullable we will set `article_id` to null in the tags table. Same for `fallback_id` you get the idea. If your tags can belong to more than one article you are using the incorrect relation should be `BelongsToMany` or `MorphToMany`

The reasoning to make it false by default, is that changing the value to "null" it's a developer feature (what is expected is that you have a table with "related keys" set so `article_id` really references an article) and people expecting this but didn't make the column non-nullable or didn't setup foreignKeys would end-up with "dead entries" in the database probably without noticing it.
Developers aware of it and that want to deal with those entries can setup the `force_delete => false` (if the field is nullable) otherwise `fallback_id => 0` for example for non-nullable columns.

I think it makes more sense, but you know... I migth be wrong!

Let me know if I expressed myself in an understandable manner! 👍

Reply:
Wait wait wait. My example doesn't make sense, nor does yours 😅  Let's use realistic examples, I think that's the source of confusion there:
- over here we're talking about `HasMany` and `MorphMany` (1-n);
- my example showed Articles & Tags, which is a bad example, because realistically between Article and Tag the relationship would be `belongsToMany` (n-n)
- your example with [Article `hasMany` Tags] is a bad example too, it's difficult to follow because never in a db will [Article `belongsTo` Tag]; 

So I'm going to try to rewrite your answer above from [Article `hasMany` Tags] to [Author  `hasMany` Books] and [Book `belongsTo` Author] (Article becomes Author, Tag becomes Book). Hopefully that way I can understand 😅 

>So, you have an Author -> HasMany -> Books | Book -> BelongsTo -> Author (via author_id in `books` table).

> If you are editing an Author you have a select for books. When those books get "unselected" they get deleted. If `force_delete = false` and column allows nullable we will set `author_id` to null in the books table. Same for `fallback_id` you get the idea. If your books can belong to more than one author you are using the incorrect relation should be `BelongsToMany` or `MorphToMany`

> The reasoning to make it false by default, is that changing the value to "null" it's a developer feature (what is expected is that you have a table with "related keys" set so `author_id` really references an article) and people expecting this but didn't make the column non-nullable or didn't setup foreignKeys would end-up with "dead entries" in the database probably without noticing it.
> Developers aware of it and that want to deal with those entries can setup the `force_delete => false` (if the field is nullable) otherwise `fallback_id => 0` for example for non-nullable columns.

> I think it makes more sense, but you know... I might be wrong!

Reply:
OK so I think I understand now, and from rephrasing the example above we're on the same page. I see a few tricky issues here, and we should be very _very_ careful about them. I'll add them in different places in this review, though, so that we can talk about them individually.

Here, I'll continue with my opinion for the `force_delete` default: **by default, it should be as non-destructive as possible**. Yes, the developer might have made a mistake when setting his column to `nullable`. But it's _safer_ to assume they did know what they were doing:
- [ ] if column is `nullable`, it MUST become whatever the `default` is on that relationship or db column... if there is no default, then it will become `null`; the user can easily override this default behaviour, using `force_delete => true` or `fallback_id => 141`; let's default to non-destructive! we can deal with orphaned entries in the docs with a big fat notice, since it's a problem... but not a big one; the dev can easily delete all orphaned entries in one SQL line, 10 years after they created the admin panel - no harm was done, just bloat which is easily identifiable;
- [ ] if column is `NOT NULL`, it should _probably_ be deleted by default; but we're still not 100% sure about that; both developer and admin should be cautioned that this will delete something - in a way they cannot miss it; THEN we can be sure we can delete the related entry; but how can we caution them?
    - [ ] for the admin
         - (A) every time the X is clicked, we could show an alert saying "_Are you sure? After you save this form, that related entry will be deleted._"; but is there a _simple_ way to add that alert, that will not bloat up our existing fields? or...
         - (B) any time `force_delete => true` we add a hint to that field (maybe yellow with `text-warning`), saying "_ATTENTION: Any entries you remove here will be deleted after you submit this form._"; this would probably be easier to implement, and make it easy for the dev to override (just change the hint); but warning hints only cover maybe 70% of all people... I know for a fact... I see it everyday in the Backpack "free license" form; people just don't read it, even if it's red;
    - [ ] for the developer, there is only one way to be certain they know the related entry will be deleted... and that's to _only do it if they tell us_; if we set `force_delete` to `false` by default, and only delete if the dev explicitly told us to delete, my making it `true`; if you haven't set that to true... when you test it... it will probably try to set null on a non-nullable column and error; this has one downside... the column won't "_just work_", you'll have to define `force_delete => true` for it to work in this particular scenario;

What do you think in terms of UI and DI @pxpm ? Personally I think:
- [ ] the yellow hint MUST be added if `force_delete` is true; it will let both the dev and most admins know what will happen;
- [ ] the alert probably SHOULD be added, to make absolutely sure admins don't delete stuff by mistake;
- [ ] the `force_delete` COULD be `false` by default; though... if we add the yellow hint... the dev will definitely notice it when they test it so... 🤷‍♂️ maybe this one is actually a WON'T;

What do you think?

Reply:
@tabacitu I know we probably gona discuss this today, but since I read your reply now and don't want to forget what I thougth in a few hours I will leave it written here:
- I am quite a bit convinced that you are right in terms of making `force_delete => false` by default.
- I totally don't agree with notices or alerts on panel. Reasoning:
     - This is a developer configuration, not a "user should be aware of" feature. 1st developer creates the panel with their rules and only after user is going to use the panel. If developer set it up to be deleted there is no need to "warn" the user that the entry is going to be deleted, that's how developer configured the panel and you know `Devs know better!` 🙃

Reply:
Yeah... ok... I guess if we do `'force_delete' => false` by default, in order for the dev to make this delete stuff they have to read the comment or docs. And in there we can make it super-clear to them what using it will entail.

Deal.